cupyx/scipy/ndimage/_filters_generic.py

```diff
@@ -0,0 +1,533 @@
+import re
+import types
+
+import cupy
+
+from cupyx.scipy.ndimage import _filters_core
+
+
+def _get_sub_kernel(f, filter_size=0, in_dtype=cupy.dtype(float)):
+    """
+    Takes the "function" given to generic_filter and returns the "sub-kernel"
+    that will be called, one of RawKernel, ReductionKernel, or FusedKernel.
+
+    This supports:
+     * cupy.RawKernel
+       no checks are possible
+     * cupy.ReductionKernel
+       checks that there is a single input and output
+     * cupy.core._fusion_kernel.FusedKernel
+       checks that there is a single input and output and some of the
+       properties of those arguments
+     * cupy.core.fusion.Fusion and cupy.core.new_fusion.Fusion
+       retrieves underlying kernel and sends it back through this function
+     * any callable
+       attempts to fuse it then and sends it back through this function
+
+    The filter_size and in_dtype arguments are the size and data type used for
+    function fusions which requires calling the fused function with an array of
+    the appropriate size and data type to create the underlying kernel.
+    """
+    if isinstance(f, cupy.RawKernel):
+        # We will assume that it has the correct API
+        return f
+    elif isinstance(f, cupy.ReductionKernel):
+        if f.nin != 1 or f.nout != 1:
+            raise TypeError('ReductionKernel must have 1 input and output')
+        return f
+    elif isinstance(f, cupy.ElementwiseKernel):
+        # special error message for ElementwiseKernels
+        raise TypeError('only ReductionKernel allowed (not ElementwiseKernel)')
+    elif isinstance(f, cupy.core._fusion_kernel.FusedKernel):
```

Comment:
I think we can remove fused kernels from this PR and add a follow-up or just for this feature.
I think there is a lot to discuss with the fusion that might need changes in the fusion interface too.

Reply:
We have been discussing here that some of the Fusion internal details are private and the only class that is accessible when a user creates a fused kernel is `cupy.core.fusion.Fusion` so this might help a bit to simplify the fusion handling

Reply:
Okay, will do that tomorrow.